src/shared/components/ui/TimeField/TimeField.tsx

```diff
@@ -0,0 +1,73 @@
+import type {
+  TimeFieldProps as AriaTimeFieldProps,
+  TimeValue,
+} from "react-aria-components";
+import {
+  TimeField as AriaTimeField,
+  DateInput,
+  DateSegment,
+  Label,
+} from "react-aria-components";
+import { FieldValues } from "react-hook-form";
+import { tv } from "tailwind-variants";
+
+import {
+  FieldMessage,
+  isFirstChild,
+  useFieldController,
+  WithHookForm,
+  WithoutHookForm,
+} from "~/shared";
+
+const classNames = tv({
+  slots: {
+    dateInput:
+      "flex w-full items-center space-x-1 rounded-lg border p-2 md:w-fit",
+    dateSegment: "font-medium text-gray-800",
+  },
+})();
```

Comment:
This was a suggestion by @slagomarsino-dev to group the styles into a `classNames` namespace and avoid confusion in the template because in this case names like `dateInput` or `dateSegment` might be confused with something else.

I agree with that but I'm also ok with calling them in the template, I think using them inside the `classNames` prop usually it's a good enough indication that it will output a class.